storage/signature.go

```diff
@@ -121,15 +154,39 @@ func (s *Signature) Open(ctx context.Context, fp string) (fs.File, error) {
 			return nil, xerrors.Errorf("read signature manifest: %w", err)
 		}
 
-		signed, err := extensionsign.SignAndZipManifest(s.Signer, manifestData)
+		vsix, err := s.Storage.Open(ctx, filepath.Join(filepath.Dir(fp), vsixPath+".vsix"))
+		if err != nil {
+			// If this file is missing, it means the extension was added before
+			// signatures were handled by the marketplace.
+			// TODO: Generate the sig manifest payload and insert it?
+			return nil, xerrors.Errorf("open signature manifest: %w", err)
+		}
+		defer vsix.Close()
+
+		vsixData, err := io.ReadAll(vsix)
+		if err != nil {
+			return nil, xerrors.Errorf("read signature manifest: %w", err)
+		}
+
+		// TODO: Fetch the VSIX payload from the storage
+		signed, err := s.SigZip(ctx, vsixData, manifestData)
 		if err != nil {
 			return nil, xerrors.Errorf("sign and zip manifest: %w", err)
```

Comment:
Oops meant to say before submitting, really big nit, but looks like this error would end up saying `sign and zip manifest: sign and zip manifest: whatever the error is` twice which could look weird